### PR TITLE
Feature/atr 548 dev check bql type compatible with property type

### PR DESCRIPTION
**Pull request by SENya1990**
Implemented PX1068 diagnostic that checks DAC fields for mismatch between property and BQL field types.

 **Changes Overview**

**PX1068 Diagnostic**
- implemented PX1068 diagnostic
- implemented PX1068 code fix that will change BQL field type into a type consistent with a property type
- implemented PX1068 code fix that will change property type into a type consistent with a BQL field type
- added unit tests for PX1068 diagnostic and code fixes
- added documentation for PX1068

**DAC Semantic Model**
- enhanced unit tests for DAC semantic model collection from DACs and DAC extensions with checks for consistency between BQL field info and DAC field info
- extended DAC field info with information about BQL field data type
- renamed `EffectivePropertyType` into `PropertyTypeUnwrappedNullable` in DAC field info and BQL field info to avoid confusion with the semantic of Declared/Effective in overridable infos
- fixed bug in DAC fields collection from DAC extensions

**Common Utilities**
- updated and renamed mapping helper between property type and BQL field type
- added `NullIfLocationKindIsNone` extension method that will correctly handle possible "null-object" locations obtained from `SyntaxToken` tokens. Integrated changes. 
- minor refactorings

### CODE REVIEW

File: docs/Summary.md [hunk 1]
```diff
@@ -75,6 +75,7 @@ Acuminator does not perform static analysis of projects whose names contain `Tes
 | [PX1065](diagnostics/PX1065.md) | The DAC field property does not have a corresponding BQL field. | Error | Available |
 | [PX1066](diagnostics/PX1066.md) | The name of the BQL field possibly contains mistake. | Warning (ISV Level 3: Informational) | Available |
 | [PX1067](diagnostics/PX1067.md) | DAC does not contain a redeclaration of a BQL field declared in the base DAC. | Warning (ISV Level 3: Informational) | Available |
+| [PX1068](diagnostics/PX1068.md) | The type of the DAC field property does not correspond to the type of the BQL field. | Error | Available |
```
Comment (SENya1990) [inline docs/Summary.md:78]: @EAndrosova  please review PX1068 summary

File: docs/diagnostics/PX1068.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# PX1068
+This document describes the PX1068 diagnostic.
```
Comment (SENya1990) [inline docs/diagnostics/PX1068.md:2]: @EAndrosova  please review PX1068 documentation

File: src/Acuminator/Acuminator.Analyzers/DiagnosticsShortName.resx [hunk 3]
```diff
@@ -339,6 +339,9 @@
   <data name="PX1067" xml:space="preserve">
     <value>MissingBqlFieldRedeclarationInDerivedDac</value>
   </data>
+  <data name="PX1068" xml:space="preserve">
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/DiagnosticsShortName.resx:342]: @EAndrosova  please review PX1068 short name

File: src/Acuminator/Acuminator.Analyzers/Resources.resx [hunk 4]
```diff
@@ -615,6 +615,15 @@ public virtual int? SomeDacField{ get; set; }</value>
   <data name="PX1067TitleFormat" xml:space="preserve">
     <value>BQL fields from base DACs should be redeclared in the derived DAC. DAC "{0}" does not contain a redeclaration of BQL field "{1}" declared in the base DAC "{2}".</value>
   </data>
+  <data name="PX1068FixBqlType" xml:space="preserve">
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/Resources.resx:618]: @EAndrosova  please review PX1068 diagnostic and code fix messages

**Review by dropsonic — APPROVED** [review]

**Review by EAndrosova — APPROVED** [review]

**Pull request merged**